dist/tools/cppcheck/check.sh

```diff
@@ -50,7 +51,30 @@ if [ -z "${FILES}" ]; then
     exit
 fi
 
+github_annotate_setup
+
 # TODO: switch back to 8 jobs when/if cppcheck issue is resolved
 cppcheck --std=c99 --enable=style --force --error-exitcode=2 --quiet -j 1 \
          --template "{file}:{line}: {severity} ({id}): {message}"         \
-         --inline-suppr ${DEFAULT_SUPPRESSIONS} ${CPPCHECK_OPTIONS} ${@} ${FILES}
+         --inline-suppr ${DEFAULT_SUPPRESSIONS} ${CPPCHECK_OPTIONS} ${@}  \
+         ${FILES} 2>&1 | ${LOG} 1>&2
+
+RESULT=$?
+
+grep "^.\+:[0-9]\+: [a-z]\+ (.*):" ${LOGFILE} | while read error_line; do
+    FILE=$(echo "${error_line}" | cut -d: -f1)
+    LINENUM=$(echo "${error_line}" | cut -d: -f2)
+    SEVERITY=$(echo "${error_line}" | cut -d: -f3)
+    ID=$(echo "${SEVERITY}" | sed 's/^.*(\(.*\))$/\1/g')
+    DETAILS=$(echo "${error_line}" | cut -d: -f4- |
+              sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
+
+    if echo "${SEVERITY}" | grep -q "\<error\>"; then
+        github_annotate_error "${FILE}" "${LINENUM}" "${ID}: ${DETAILS}"
```

Comment:
The only severity level that justifies an error annotation is "error", see [cppcheck doc](http://cppcheck.sourceforge.net/devinfo/doxyoutput/group__Core.html#gac185938ae084355bbf1790cf1a70caa6). Not sure if this is confusing though, since I am planning to use `github_annotate_warn` for the warning coccinelle checks and other more softer chechs e.g... In the end, I guess a newcomer will diligently fix all issues pointed out by the annotations, so maybe this isn't that confusing after all, the warnings should still be addressed, right? (except maybe those 80 chars / line thingies by vera++ ;-P)